pyne/xs/models.pyx

```diff
@@ -201,16 +224,25 @@ def group_collapse(sigma_n, phi_n, phi_g=None, partial_energies=None, E_g=None,
         pem = partial_energies
     elif (phi_g is None) and (partial_energies is not None):
         pem = partial_energies
-        phi_g = np.dot(pem, phi_n)
+        if wgts == None:
+           phi_g = np.dot(pem, phi_n)
+        else:
+           phi_g = np.dot(pem, phi_n * wgts[::-1])
     elif (E_g is not None) and (E_n is not None):
         pem =  partial_energy_matrix(E_g, E_n)
-        phi_g = np.dot(pem, phi_n)
+        if wgts == None:
+           phi_g = np.dot(pem, phi_n)
+        else:
+           phi_g = np.dot(pem, phi_n * wgts[::-1])
```

Comment:
the `weights[::-1]` thing seems weird to me.  Shouldn't we just enforce / assume that the weights have the same tonicity as `phi_n`, not enforce that it has the opposite?